main/src/main/java/cgeo/geocaching/unifiedmap/layers/GeoItemsLayer.java

```diff
@@ -32,16 +34,22 @@ public GeoItemsLayer(final AppCompatActivity activity, final GeoItemLayer<String
 
             final Map<String, Integer> currentlyDisplayedGeocaches = new HashMap<>();
 
+            final boolean forceCompactIconMode = CompactIconModeUtils.forceCompactIconMode(caches.size()); // todo: only use caches in viewport and update while moving the map
```

Comment:
We need to count caches in viewport anyway, to be able to update the cache counter in toolbar, so maybe this can be put into the viewmodel cache observer, updating GeoItemLayer and holding a counter available for UnifiedMapActivity for triggering toolbar update after filtering?

Reply:
We IMHO need to observe map movements directly, as the cache list is only refreshed if necessary. This is however not yet designed in our map specific implementations.